rumqttd/src/router/routing.rs

```diff
@@ -946,6 +941,9 @@ impl Router {
                 .insert(filter_path.clone(), subscription_id);
         }
 
+        let forward_retained_msg =
+            (retain_forward_rule != &protocol::RetainForwardRule::Never) && group.is_none();
```

Comment:
why do we check `group.is_none()` here?

Reply:
AFAIK, retained messages are only sent for normal subscriptions, not shared subscriptions